src/main.rs

```diff
@@ -232,11 +300,214 @@ mod cli {
             Kanata::start_notification_loop(nrx, server.connections);
         }
 
+        // Start file watcher if enabled
+        #[cfg(feature = "watch")]
+        if args.watch {
+            if let Err(e) = file_watcher::start_file_watcher(args.paths.clone(), kanata_arc.clone()) {
+                log::error!("Failed to start file watcher: {}", e);
+            }
+        }
+
         #[cfg(target_os = "linux")]
         sd_notify::notify(true, &[sd_notify::NotifyState::Ready])?;
 
         Kanata::event_loop(kanata_arc, tx)
     }
+
+    #[cfg(all(feature = "watch", test))]
+    mod tests {
+        use super::*;
+        use std::fs;
+        use tempfile::NamedTempFile;
+
+        #[test]
+        fn test_request_live_reload_method() {
+            // Test that the new request_live_reload method works
+            let temp_file = NamedTempFile::with_suffix(".kbd").unwrap();
+            let config_path = temp_file.path().to_path_buf();
+            
+            // Write a minimal valid config
+            fs::write(&config_path, r#"
+                (defsrc caps)
+                (deflayer default caps)
+            "#).unwrap();
+
+            let args = ValidatedArgs {
+                paths: vec![config_path],
+                #[cfg(feature = "tcp_server")]
+                tcp_server_address: None,
+                #[cfg(target_os = "linux")]
+                symlink_path: None,
+                nodelay: true,
+                #[cfg(feature = "watch")]
+                watch: false,
+            };
+
+            let mut kanata = Kanata::new(&args).unwrap();
+            
+            // Initially, live_reload_requested should be false
+            assert!(!kanata.is_live_reload_requested());
+            
+            // Request live reload
+            kanata.request_live_reload();
+            
+            // Now it should be true
+            assert!(kanata.is_live_reload_requested());
+        }
+
+        #[test]
+        fn test_file_watcher_path_validation() {
```

Comment:
If I'm reading correctly, this doesn't seem to be testing any Kanata code. If I am correct, remove this test.